### ISSUE TITLE
[Feature]: Add Paired Region information to get-azlocation cmdlet

### ISSUE THREAD

**Issue by brwilkinson** [Azure/azure-powershell#17536]
### Description of the new feature

https://docs.microsoft.com/en-us/rest/api/resources/subscriptions/list-locations

`GET https://management.azure.com/subscriptions/291bba3f-e0a5-47bc-a099-3bdcb2a50a05/locations?api-version=2020-01-01`

```json
  {
    "id": "/subscriptions/291bba3f-e0a5-47bc-a099-3bdcb2a50a05/locations/centralus",
    "name": "centralus",
    "displayName": "Central US",
    "regionalDisplayName": "(US) Central US",
    "metadata": {
      "regionType": "Physical",
      "regionCategory": "Recommended",
      "geographyGroup": "US",
      "longitude": "-93.6208",
      "latitude": "41.5908",
      "physicalLocation": "Iowa",
      "pairedRegion": [
        {
          "name": "eastus2",
          "id": "/subscriptions/291bba3f-e0a5-47bc-a099-3bdcb2a50a05/locations/eastus2"
        }
      ]
    }
  }
```

### Proposed implementation details (optional)

update from

Absolute Uri:
https://management.azure.com/subscriptions/b8f402aa-20f7-4888-b45c-3cf086dad9c3/locations?api-version=2016-06-01

to
https://management.azure.com/subscriptions/291bba3f-e0a5-47bc-a099-3bdcb2a50a05/locations?api-version=2020-01-01

Then add the extra property.

```powershell
get-azlocation | gm

   TypeName: Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels.PSResourceProviderLocation

Name        MemberType Definition
----        ---------- ----------
Equals      Method     bool Equals(System.Object obj)
GetHashCode Method     int GetHashCode()
GetType     Method     type GetType()
ToString    Method     string ToString()
DisplayName Property   string DisplayName {get;set;}
Location    Property   string Location {get;set;}
Providers   Property   System.Collections.Generic.List[string] Providers {get;set;}
```

from:

```powershell
get-azlocation | select -first 1

Location    : eastasia
DisplayName : East Asia
Providers   : {microsoft.insights, Microsoft.OperationalInsights, Microsoft.OperationsManagement, Microsoft.ContainerRegistry…}
```

to 

```powershell
get-azlocation | select -first 1

Location    : eastasia
DisplayName : East Asia
pairedRegion: {pairedRegionInfo}
Providers   : {microsoft.insights, Microsoft.OperationalInsights, Microsoft.OperationsManagement, Microsoft.ContainerRegistry…}
```

**Comment by ArmaanMcleod** [comment]
Would also add that forcing the Api version with `Get-AzLocation -ApiVersion `2020-01-01` doesn't change anything either. This is probably due to the `SubscriptionClientWrapper` setting the API version to `2016-06-01`. I would imagine to fix this you would need to upgrade this API version to `2020-01-01` then map the paired regions from the metadata to a member in `PSResourceProviderLocation`. Might also need to make this cmdlet work with different API versions if that is how its meant to work. I don't really see why anyone would want to use an older api version though.

**Comment by ksaffarian** [comment]
Hi @Nickcandy @dingmeng-xue 
Is this fixed? I could not understand why it is marked as completed.
I am seeing this issue in Azure DevOps when using AzurePowerShell tasks both v4 and v5 (https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/azure-powershell?view=azure-devops)

I took my hair out until I found this page. My script works locally, when reading `pairedRegion` but it fails when running it in Azure DevOps.

Thanks
